lib/api_entity.rb

```diff
@@ -140,13 +140,16 @@ def has_one(association, opts = {})
       attr_reader association.to_sym
 
       define_method("#{association}=") do |attributes|
-        class_name = if options[:polymorphic]
-                       attributes['resource_type'].classify
-                     else
-                       options[:class_name]
-                     end
-
-        entity = class_name.constantize.new((attributes.presence || {}).merge(casted_by: self))
+        entity = if attributes.nil?
```

Comment:
This looks like a change of behaviour - previously a relationship without attributes (which is valid I think) - though initialize `options[:class_name]` with an empty hash (in the non polymorphic scenario)

Now it sets the relationship to nil instead? What if the relationship entity is just a join model? We have that scenario in permutations?

Reply:
It is definitely a change in behaviour. We're going from not supporting (exploding) on null entities, to supporting null entities